pkg/config/config.go

```diff
@@ -671,6 +671,7 @@ func InitConfig(config Config) {
 	config.SetKnown("proxy.http")
 	config.SetKnown("proxy.https")
 	config.SetKnown("proxy.no_proxy")
+	config.SetKnown("proxy.no_proxy_nonexact_match")
```

Comment:
We need to use `BindEnvAndSetDefault` so the env var `DD_PROXY_NO_PROXY_NONEXACT_MATCH` is supported.